units/src/androidMain/kotlin/dev/icerock/moko/units/databinding/BindingAdapters.kt

```diff
@@ -47,7 +48,12 @@ fun RecyclerView.setAdapter(adapterClassName: String) {
     val adapterClass = this.javaClass.classLoader?.loadClass(adapterClassName)
 
     if (adapterClass == UnitsRecyclerViewAdapter::class.java) {
-        adapter = UnitsRecyclerViewAdapter(context as LifecycleOwner)
+        val lifecycleOwner = if (context is ContextThemeWrapper) {
+            (context as ContextThemeWrapper).baseContext as LifecycleOwner
```

Comment:
here `context` is already smart-casted to `ContextThemeWrapper` so you can simple `context.baseContext as LifecycleOwner`

Reply:
In this scope `context` is a getter so we need to put it in val before type check.